firestore/integration_test_internal/src/transaction_test.cc

```diff
@@ -743,21 +745,24 @@ TEST_F(TransactionTest, TestMaxAttempts) {
 
   Future<void> run_transaction_future = firestore->RunTransaction(
       options,
-      [update_count, &doc](Transaction& transaction, std::string&) -> Error {
-        SCOPED_TRACE("Update callback; update_count=" +
-                     std::to_string(update_count->load()));
+      [update_count, &doc](Transaction& transaction,
+                           std::string& error_message) -> Error {
+        SCOPED_TRACE(absl::StrCat("Update callback; update_count=",
+                                  update_count->load()));
         ++(*update_count);
 
         // Get the document via the transaction.
         {
           SCOPED_TRACE("transaction.Get()");
           Error error = Error::kErrorOk;
-          std::string error_message;
-          transaction.Get(doc, &error, &error_message);
+          std::string get_error_message;
+          transaction.Get(doc, &error, &get_error_message);
           if (error != kErrorOk) {
-            ADD_FAILURE() << "transaction.Get() failed: " << error_message
-                          << " (error code " << error << ")";
-            return Error::kErrorInternal;
+            ADD_FAILURE() << "transaction.Get() failed at Checkpoint UCA: "
+                          << get_error_message << " (error code " << error
+                          << "==" << ToFirestoreErrorCodeName(error) << ")";
+            error_message = "Test failed in update callback at Checkpoint UCA";
+            return Error::kErrorInvalidArgument;
```

Comment:
Note: I changed kErrorInternal -> kErrorInvalidArgument because the latter is considered to be a "permanent error" and will prevent the transaction from being uselessly, and confusingly, retried.